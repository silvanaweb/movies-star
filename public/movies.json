{
	"results": [
		{
			"vote_count": 214,
			"id": 456740,
			"video": false,
			"vote_average": 5.2,
			"title": "Hellboy",
			"popularity": 375.676,
			"poster_path": "/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg",
			"original_language": "en",
			"original_title": "Hellboy",
			"genre_ids": [28, 12, 14],
			"backdrop_path": "/5BkSkNtfrnTuKOtTaZhl8avn4wU.jpg",
			"adult": false,
			"overview": "Hellboy comes to England, where he must defeat Nimue, Merlin's consort and the Blood Queen. But their battle will bring about the end of the world, a fate he desperately tries to turn away.",
			"release_date": "2019-04-10"
		},
		{
			"vote_count": 4031,
			"id": 299537,
			"video": false,
			"vote_average": 7.1,
			"title": "Captain Marvel",
			"popularity": 322.559,
			"poster_path": "/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
			"original_language": "en",
			"original_title": "Captain Marvel",
			"genre_ids": [28, 12, 878],
			"backdrop_path": "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
			"adult": false,
			"overview": "The story follows Carol Danvers as she becomes one of the universe’s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races. Set in the 1990s, Captain Marvel is an all-new adventure from a previously unseen period in the history of the Marvel Cinematic Universe.",
			"release_date": "2019-03-06"
		},
		{
			"vote_count": 1155,
			"id": 287947,
			"video": false,
			"vote_average": 7.2,
			"title": "Shazam!",
			"popularity": 304.941,
			"poster_path": "/xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
			"original_language": "en",
			"original_title": "Shazam!",
			"genre_ids": [35, 12, 14],
			"backdrop_path": "/bi4jh0Kt0uuZGsGJoUUfqmbrjQg.jpg",
			"adult": false,
			"overview": "A boy is given the ability to become an adult superhero in times of need with a single magic word.",
			"release_date": "2019-03-23"
		},
		{
			"vote_count": 463,
			"id": 537915,
			"video": false,
			"vote_average": 6.5,
			"title": "After",
			"popularity": 288.044,
			"poster_path": "/u3B2YKUjWABcxXZ6Nm9h10hLUbh.jpg",
			"original_language": "en",
			"original_title": "After",
			"genre_ids": [18, 10749],
			"backdrop_path": "/8lESY7UGpOsbL2caib9Qe4bOebF.jpg",
			"adult": false,
			"overview": "A young woman falls for a guy with a dark secret and the two embark on a rocky relationship.",
			"release_date": "2019-04-11"
		},
		{
			"vote_count": 1548,
			"id": 166428,
			"video": false,
			"vote_average": 7.6,
			"title": "How to Train Your Dragon: The Hidden World",
			"popularity": 164.927,
			"poster_path": "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
			"original_language": "en",
			"original_title": "How to Train Your Dragon: The Hidden World",
			"genre_ids": [16, 10751, 12],
			"backdrop_path": "/lFwykSz3Ykj1Q3JXJURnGUTNf1o.jpg",
			"adult": false,
			"overview": "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.",
			"release_date": "2019-01-03"
		},
		{
			"vote_count": 98,
			"id": 480414,
			"video": false,
			"vote_average": 5.7,
			"title": "The Curse of La Llorona",
			"popularity": 149.814,
			"poster_path": "/jhZlXSnFUpNiLAek9EkPrtLEWQI.jpg",
			"original_language": "en",
			"original_title": "The Curse of La Llorona",
			"genre_ids": [53, 27, 9648],
			"backdrop_path": "/u2vGggeMPAkhEtD7bYGfeThsQiM.jpg",
			"adult": false,
			"overview": "A social worker dealing with the disappearance of two children fears for her own family after beginning the investigation.",
			"release_date": "2019-04-17"
		},
		{
			"vote_count": 937,
			"id": 329996,
			"video": false,
			"vote_average": 6.7,
			"title": "Dumbo",
			"popularity": 116.126,
			"poster_path": "/279PwJAcelI4VuBtdzrZASqDPQr.jpg",
			"original_language": "en",
			"original_title": "Dumbo",
			"genre_ids": [12, 10751, 14],
			"backdrop_path": "/5tFt6iuGnKapHl5tw0X0cKcnuVo.jpg",
			"adult": false,
			"overview": "A young elephant, whose oversized ears enable him to fly, helps save a struggling circus, but when the circus plans a new venture, Dumbo and his friends discover dark secrets beneath its shiny veneer.",
			"release_date": "2019-03-27"
		},
		{
			"vote_count": 2023,
			"id": 424783,
			"video": false,
			"vote_average": 6.4,
			"title": "Bumblebee",
			"popularity": 115.64,
			"poster_path": "/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
			"original_language": "en",
			"original_title": "Bumblebee",
			"genre_ids": [28, 12, 878],
			"backdrop_path": "/wtZj5nn6hVwgakPdg6y6gm3eFXU.jpg",
			"adult": false,
			"overview": "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.",
			"release_date": "2018-12-15"
		},
		{
			"vote_count": 1117,
			"id": 458723,
			"video": false,
			"vote_average": 7.2,
			"title": "Us",
			"popularity": 113.813,
			"poster_path": "/ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
			"original_language": "en",
			"original_title": "Us",
			"genre_ids": [53, 27],
			"backdrop_path": "/jNUCddkM1fjYcFIcEwFjc7s2H4V.jpg",
			"adult": false,
			"overview": "Husband and wife Gabe and Adelaide Wilson take their kids to their beach house expecting to unplug and unwind with friends. But as night descends, their serenity turns to tension and chaos when some shocking visitors arrive uninvited.",
			"release_date": "2019-03-14"
		},
		{
			"vote_count": 909,
			"id": 522681,
			"video": false,
			"vote_average": 6.3,
			"title": "Escape Room",
			"popularity": 103.399,
			"poster_path": "/8Ls1tZ6qjGzfGHjBB7ihOnf7f0b.jpg",
			"original_language": "en",
			"original_title": "Escape Room",
			"genre_ids": [53],
			"backdrop_path": "/mWLljCmpqlcYQh7uh51BBMwCZwN.jpg",
			"adult": false,
			"overview": "Six strangers find themselves in circumstances beyond their control, and must use their wits to survive.",
			"release_date": "2019-01-03"
		},
		{
			"vote_count": 144,
			"id": 487297,
			"video": false,
			"vote_average": 5.5,
			"title": "What Men Want",
			"popularity": 91.606,
			"poster_path": "/30IiwvIRqPGjUV0bxJkZfnSiCL.jpg",
			"original_language": "en",
			"original_title": "What Men Want",
			"genre_ids": [35, 14, 10749],
			"backdrop_path": "/umecegsPpKr2ZXA62Da9CQBVoIO.jpg",
			"adult": false,
			"overview": "Magically able to hear what men are thinking, a sports agent uses her newfound ability to turn the tables on her overbearing male colleagues.",
			"release_date": "2019-02-08"
		},
		{
			"vote_count": 1,
			"id": 498580,
			"video": false,
			"vote_average": 0,
			"title": "Kanchana 3",
			"popularity": 87.69,
			"poster_path": "/eV5uk58nNkTIDyn0HF9Hio1YBLX.jpg",
			"original_language": "ta",
			"original_title": "Kanchana 3",
			"genre_ids": [35, 27],
			"backdrop_path": "/6IftJA2bm4IBEGJV1nG7oJzK5TB.jpg",
			"adult": false,
			"overview": "Kanchana 3 is an UpcomingTamil horror Revenge film directed by Raghava Lawrence, starring himself, Oviya and Vedhika in the leading roles.",
			"release_date": "2019-04-19"
		},
		{
			"vote_count": 1293,
			"id": 504172,
			"video": false,
			"vote_average": 6.4,
			"title": "The Mule",
			"popularity": 76.285,
			"poster_path": "/oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
			"original_language": "en",
			"original_title": "The Mule",
			"genre_ids": [80, 18, 53],
			"backdrop_path": "/bTeRgkAavyw1eCtSkaww18wLYNP.jpg",
			"adult": false,
			"overview": "Earl Stone, a man in his 80s, is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates.",
			"release_date": "2018-12-14"
		},
		{
			"vote_count": 107,
			"id": 471507,
			"video": false,
			"vote_average": 5.7,
			"title": "Destroyer",
			"popularity": 71.728,
			"poster_path": "/sHw9gTdo43nJL82py0oaROkXXNr.jpg",
			"original_language": "en",
			"original_title": "Destroyer",
			"genre_ids": [53, 80, 18, 28],
			"backdrop_path": "/rURV5xPzt9ZxEXAc4OQmxchGyZ8.jpg",
			"adult": false,
			"overview": "Erin Bell is an LAPD detective who, as a young cop, was placed undercover with a gang in the California desert with tragic results. When the leader of that gang re-emerges many years later, she must work her way back through the remaining members and into her own history with them to finally reckon with the demons that destroyed her past.",
			"release_date": "2018-12-25"
		},
		{
			"vote_count": 62,
			"id": 411728,
			"video": false,
			"vote_average": 7.3,
			"title": "The Professor and the Madman",
			"popularity": 67.699,
			"poster_path": "/gtGCDLhfjW96qVarwctnuTpGOtD.jpg",
			"original_language": "en",
			"original_title": "The Professor and the Madman",
			"genre_ids": [18, 36, 9648],
			"backdrop_path": "/zhNfM1CSQqJtIqelwlK9iKtSd9P.jpg",
			"adult": false,
			"overview": "Professor James Murray begins work compiling words for the first edition of the Oxford English Dictionary in the mid 19th century and receives over 10,000 entries from a patient at Broadmoor Criminal Lunatic Asylum , Dr William Minor.",
			"release_date": "2019-03-07"
		},
		{
			"vote_count": 298,
			"id": 157433,
			"video": false,
			"vote_average": 5.9,
			"title": "Pet Sematary",
			"popularity": 63.566,
			"poster_path": "/7SPhr7Qj39vbnfF9O2qHRYaKHAL.jpg",
			"original_language": "en",
			"original_title": "Pet Sematary",
			"genre_ids": [27, 53],
			"backdrop_path": "/n2aX63BmW7zIKgKJ58e6rKlSsdi.jpg",
			"adult": false,
			"overview": "Louis Creed, his wife Rachel and their two children Gage and Ellie move to a rural home where they are welcomed and enlightened about the eerie 'Pet Sematary' located nearby. After the tragedy of their cat being killed by a truck, Louis resorts to burying it in the mysterious pet cemetery, which is definitely not as it seems, as it proves to the Creeds that sometimes dead is better.",
			"release_date": "2019-04-04"
		},
		{
			"vote_count": 6963,
			"id": 424694,
			"video": false,
			"vote_average": 8.1,
			"title": "Bohemian Rhapsody",
			"popularity": 63.175,
			"poster_path": "/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg",
			"original_language": "en",
			"original_title": "Bohemian Rhapsody",
			"genre_ids": [18, 10402],
			"backdrop_path": "/xcaSYLBhmDzJ6P14bcKe0KTh3QV.jpg",
			"adult": false,
			"overview": "Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet – finding a way to keep the band together amid the success and excess.",
			"release_date": "2018-10-24"
		},
		{
			"vote_count": 137,
			"id": 454294,
			"video": false,
			"vote_average": 6.1,
			"title": "The Kid Who Would Be King",
			"popularity": 62.813,
			"poster_path": "/kBuvLX6zynQP0sjyqbXV4jNaZ4E.jpg",
			"original_language": "en",
			"original_title": "The Kid Who Would Be King",
			"genre_ids": [28, 12, 14, 10751],
			"backdrop_path": "/jmbgxBd86MshzQQvv5laKvOKoMm.jpg",
			"adult": false,
			"overview": "Old-school magic meets the modern world when young Alex stumbles upon the mythical sword Excalibur. He soon unites his friends and enemies, and they become knights who join forces with the legendary wizard Merlin. Together, they must save mankind from the wicked enchantress Morgana and her army of supernatural warriors.",
			"release_date": "2019-01-16"
		},
		{
			"vote_count": 28,
			"id": 458253,
			"video": false,
			"vote_average": 7.3,
			"title": "Missing Link",
			"popularity": 61.882,
			"poster_path": "/gEkKHiiQRVUSX15Iwo8VFydXrtu.jpg",
			"original_language": "en",
			"original_title": "Missing Link",
			"genre_ids": [16, 35, 12, 14, 10751],
			"backdrop_path": "/ob4XnKBqAh1rg381hayebJx6EkE.jpg",
			"adult": false,
			"overview": "The charismatic Sir Lionel Frost considers himself to be the world's foremost investigator of myths and monsters. Trouble is, none of his small-minded, high-society peers seems to recognize this. Hoping to finally gain acceptance from these fellow adventurers, Sir Lionel travels to the Pacific Northwest to prove the existence of a legendary creature known as the missing link.",
			"release_date": "2019-04-04"
		},
		{
			"vote_count": 422,
			"id": 500682,
			"video": false,
			"vote_average": 6.7,
			"title": "The Highwaymen",
			"popularity": 60.166,
			"poster_path": "/4bRYg4l12yDuJvAfqvUOPnBrxno.jpg",
			"original_language": "en",
			"original_title": "The Highwaymen",
			"genre_ids": [80, 18, 53],
			"backdrop_path": "/pZ78ksjPlXf3q2EeONN8WdHE03Y.jpg",
			"adult": false,
			"overview": "In 1934, Frank Hamer and Manny Gault, two former Texas Rangers, are commissioned to put an end to the wave of vicious crimes perpetrated by Bonnie Parker and Clyde Barrow, a notorious duo of infamous robbers and cold-blooded killers who nevertheless are worshiped by the public.",
			"release_date": "2019-03-15"
		}
	]
}
