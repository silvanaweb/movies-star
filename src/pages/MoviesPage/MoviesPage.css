.Movies__filter {
  margin: var(--l-size) 0;
}

.Movies__ratings__filter {
  margin: 0 auto;
}

.Movies__list__empty {
  margin-top: var(--l-size);
  font-size: var(--font-medium);
  text-align: center;
}

.Movies__results {
  font-size: var(--font-normal);
  color: var(--teal);
  margin-top: var(--m-size);
}

@media (min-width: 600px) {
  .Movies__ratings {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
  .Movies__ratings__filter {
    width: 65%;
  }
  .Movies__results {
    font-size: var(--font-medium);
  }
}
