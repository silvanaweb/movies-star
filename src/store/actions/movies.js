import { orderBy } from "lodash";

// SET_MOVIES
const setMovies = movies => ({
	type: "SET_MOVIES",
	movies,
});

// genres are mapped in the genres list of the movie,
// this  facilitates the viewving of the genres
// (alternatively this mapping could be done before rendering, but because the list does not change, it makes sense doing it here)
const mapListMovie = genresMap => ({ vote_average, title, genre_ids, poster_path, id }) => ({
	rating: vote_average,
	title,
	genres: genre_ids.map(gid => genresMap[gid]),
	poster_path,
	id,
});

export const formatMovies = (moviesRaw, genres) => {
	const genresMap = genres.reduce((acc, { id, name }) => ({ ...acc, [id]: name }), {});
	const sortedMovies = orderBy(moviesRaw, ["popularity"], ["desc"]);
	return sortedMovies.map(mapListMovie(genresMap));
};

const startSetMovies = (api, genres) => {
	return dispatch => {
		// fetch movies from DB
		return api
			.getMovies()
			.then(moviesRaw => {
				// sort the moives by popularity. this operation is performed here rather than in the selector
				// because it is supposed to be the default and only sorting operation
				// extract the required fields from the movie object (do not need to fill up the memory with unused info)
				dispatch(setMovies(formatMovies(moviesRaw, genres)));
			})
			.catch(error => {
				throw new Error(error);
			});
	};
};

export { setMovies, startSetMovies };
