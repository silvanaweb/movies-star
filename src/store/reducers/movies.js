const moviesReducerDefaultState = [];

export default (state = moviesReducerDefaultState, action) => {
  switch (action.type) {
    case "SET_MOVIES":
      return action.movies;

    default:
      return state;
  }
};
