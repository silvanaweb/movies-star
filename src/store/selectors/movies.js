export default (movies, { genres, rating }) => {
  let filteredMovies = movies;

  if (!!rating) {
    filteredMovies = filteredMovies.filter(
      movie => movie.vote_average >= rating
    );
  }

  if (!!genres.length) {
    filteredMovies = filteredMovies.filter(movie =>
      genres.every(g => movie.genres.includes(g))
    );
  }

  return filteredMovies;
};
