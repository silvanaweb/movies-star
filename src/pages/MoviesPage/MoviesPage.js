import React from "react";
import { connect } from "react-redux";
import FilterByGenre from "../../components/Filters/FilterByGenre";
import FilterByRating from "../../components/Filters/FilterByRating";
import { MovieList } from "../../components/Movies/MovieList";
import selectMovies from "../../store/selectors/movies";
import "./MoviesPage.css";

const MoviesPage = ({ movieList }) => {
  return (
    <>
      <div className="content-container">
        <FilterByGenre />
        <div className="Movies__ratings">
          <FilterByRating className="Movies__ratings__filter" />{" "}
        </div>
      </div>
      <div className="content-container">
        {!!movieList.length ? (
          <div>
            <p className="Movies__results">
              You have found {movieList.length} movies
            </p>
            <MovieList movies={movieList} />
          </div>
        ) : (
          <div className="Movies__list__empty">No movies available</div>
        )}
      </div>
    </>
  );
};

const mapStateToProps = (state, props) => {
  return {
    movieList: selectMovies(state.movies, state.filters)
  };
};

export { MoviesPage };
export default connect(mapStateToProps)(MoviesPage);
