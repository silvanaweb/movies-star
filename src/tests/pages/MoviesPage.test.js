import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { MoviesPage } from "../../pages/MoviesPage/MoviesPage";
import { movies } from "../fixtures/movies";

test("should render NotFoundPage correctly with empty list", () => {
	const wrapper = shallow(<MoviesPage movieList={[]} />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should handle onGoToHome correctly with movies list", () => {
	const wrapper = shallow(<MoviesPage movieList={movies} />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
