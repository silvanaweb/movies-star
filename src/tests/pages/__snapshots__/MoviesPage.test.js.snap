// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle onGoToHome correctly with movies list 1`] = `
<div>
  <div
    className="content-container"
  >
    <ConnectFunction />
    <div
      className="Movies__ratings"
    >
      <ConnectFunction
        className="Movies__ratings__filter"
      />
       
    </div>
  </div>
  <div
    className="content-container"
  >
    <div>
      <p
        className="Movies__results"
      >
        You have found 
        3
         movies
      </p>
      <MovieList
        movies={
          Array [
            Object {
              "genres": Array [
                "Action",
              ],
              "id": 2,
              "poster_path": "image.jpg",
              "rating": 8,
              "title": "Forrest Gump",
            },
            Object {
              "genres": Array [
                "Animation",
                "Adventure",
              ],
              "id": 3,
              "poster_path": "image.jpg",
              "rating": 5,
              "title": "Shrek",
            },
            Object {
              "genres": Array [
                "Action",
                "Adventure",
              ],
              "id": 1,
              "poster_path": "image.jpg",
              "rating": 7,
              "title": "The Last Bender",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;

exports[`should render NotFoundPage correctly with empty list 1`] = `
<div>
  <div
    className="content-container"
  >
    <ConnectFunction />
    <div
      className="Movies__ratings"
    >
      <ConnectFunction
        className="Movies__ratings__filter"
      />
       
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      className="Movies__list__empty"
    >
      No movies available
    </div>
  </div>
</div>
`;
